### ISSUE TITLE
[naming-convention] `method` selector limits ability to control format of properties (vs `memberLike`)

### ISSUE THREAD

**Issue by G-Rath** [typescript-eslint/typescript-eslint#1477]
The titles a bit average, but this relates a bit to #1475.

Basically, because of the overlap of `memberLike` & `method`, it's not possible to have allow properties that are assigned functions to have the `PascelCase` format without allowing public methods on classes to have that format too.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/naming-convention": ["error", 
       { "selector": "memberLike", "format": ["camelCase"] },
       { "selector": "property", "format": ["camelCase", "PascalCase"] }
    ]
  }
}
```

```TS
class MyClass {
  public MyMethod(): void {
    //
  }
}

const o = {
  Literal(node): void {
    //
  }
};
```

**Expected Result**
`MyClass#MyMethod` is flagged as not being camelcase, while `o#Literal` is fine.

**Actual Result**
Both `MyMethod` & `Literal` are flagged as not being in camelcase.

**Additional Info**

**Versions**

Also tried against canary.

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.16.0` |
| `@typescript-eslint/parser`        | `2.16.0` |
| `TypeScript`                       | `3.7.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.14.1` |
| `npm`                              | `6.13.6` |

**Comment by bradzacher** [comment]
Using the config you provided, it looks like it works as expected:

![image](https://user-images.githubusercontent.com/7462525/72766591-706c2680-3ba5-11ea-86a6-373b8653b7ce.png)

**Comment by G-Rath** [comment]
> Using the config you provided, it looks like it works as expected:

Except that's not what I expected:

> Expected Result
> MyClass#MyMethod is flagged as not being camelcase, **while o#Literal is fine.**

The configuration I provided was an attempt to have the Expected Result I specified, but is not necessarily the "correct" configuration, as the right solution here might be creating a new specifier (i.e "functionProperties").

I've also tried using the `types: ["function"]`, which didn't work.

**Comment by bradzacher** [comment]
Sorry, I misunderstood the original post!

--

I don't have a problem with making the rule more customisable!

Happy to have a change to convert `method` to a group selector, and then introducing `classMethod` and `objectMethod` selectors.

**Comment by G-Rath** [comment]
That was exactly what I had in mind, just with much better names 😄
